src/InstrumentationEngine/ProfilerManager.cpp

```diff
@@ -2687,10 +2799,35 @@ HRESULT CProfilerManager::ProfilerAttachComplete(void)
 {
     HRESULT hr = S_OK;
 
-    // TODO: How to handle this? Does this need to support profiler attach? How does that impact initlization etc...
-
     PROF_CALLBACK_BEGIN
 
+    // Populate module, assembly, and app domain information.
+    CComQIPtr<ICorProfilerInfo3> pProfilerInfo3 = m_pRealProfilerInfo;
+    IfNullRet(pProfilerInfo3);
+
+    CComPtr<ICorProfilerModuleEnum> pModuleEnum;
+    IfFailRet(pProfilerInfo3->EnumModules(&pModuleEnum));
+
+    bool fContinueModuleEnum;
+    do
+    {
+        ULONG ulFetched = 0;
+        ModuleID moduleIds[1];
+        IfFailRet(pModuleEnum->Next(1, moduleIds, &ulFetched));
+
+        if (fContinueModuleEnum = (S_OK == hr))
+        {
+            // Constructing the module info will also construct the assembly
+            // and app domain info associated with the module.
+            CComPtr<IModuleInfo> pModuleInfo;
+            IfFailRet(ConstructModuleInfo(moduleIds[0], &pModuleInfo));
```

Comment:
>ConstructModuleInfo [](start = 22, length = 19)

I can't remember what we discussed here, but we don't want to "replay" ModuleLoaded here correct, but rather each IM would decide how to instrument inside AttachComplete?

Reply:
That's what I'm thinking for now. Makes it easier for each IM to decide what it wants to do.

In the future, we could add a new IInstrumentationMethodAttach* interface with a method that allows the engine to ask the method if it would like the engine to replay app domain, assembly, and module loads for it. The caveat is that the IM would have to know to call ApplyMetadata during those replays on attach whereas for profiling at startup does not require the ApplyMetadata call, but that could just easily be documented (or if we want to get crafty, we could monitor the metadata emit interfaces and automatically apply the changes if changes were made).